bee-node/src/node/builder.rs

```diff
@@ -35,20 +41,41 @@ type WorkerStart<N> = dyn for<'a> FnOnce(&'a mut N) -> Pin<Box<dyn Future<Output
 type WorkerStop<N> = dyn for<'a> FnOnce(&'a mut N) -> Pin<Box<dyn Future<Output = ()> + Send + 'a>> + Send;
 type ResourceRegister<N> = dyn for<'a> FnOnce(&'a mut N);
 
-fn ctrl_c_listener() -> oneshot::Receiver<()> {
+fn shutdown_procedure(sender: oneshot::Sender<()>) {
+    warn!("Gracefully shutting down the node, this may take some time.");
+
+    if let Err(e) = sender.send(()) {
+        panic!("Failed to send the shutdown signal: {:?}", e);
+    }
+}
+
+fn shutdown_listener(#[cfg(unix)] signals: Vec<SignalKind>) -> oneshot::Receiver<()> {
     let (sender, receiver) = oneshot::channel();
 
-    tokio::spawn(async move {
-        if let Err(e) = tokio::signal::ctrl_c().await {
-            panic!("Failed to intercept CTRL-C: {:?}.", e);
-        }
+    cfg_if! {
```

Comment:
I don't know if adding a new crate to do this is worth it. Do you think that is there a downside of just having a `shutdown_listener` function for `#[cfg(unix)]` and another one for `#[cfg(not(unix))]` instead?

Reply:
Nope, it just seemed to make things a little nicer. No problems removing it, though.

Reply:
Removed.